scvi/train/_metrics.py

```diff
@@ -10,64 +10,81 @@ class ElboMetric(Metric):
 
     Parameters
     ----------
-    n_obs_total
-        Number of total observations, for rescaling the ELBO
+    name
+        Name of metric
     mode
         Train or validation, used for logging names
+    interval
+        The interval over which the metric is computed. If "obs", the metric value
+        per observation is computed. If "batch", the metric value per batch is computed.
     dist_sync_on_step
         optional, by default False
     **kwargs
         Keyword args for :class:`torchmetrics.Metric`
     """
 
+    _N_OBS_MINIBATCH_KEY = "n_obs_minibatch"
+
     def __init__(
         self,
-        n_obs_total: int,
+        name: str,
         mode: Literal["train", "validation"],
+        interval: Literal["obs", "batch"],
         dist_sync_on_step: bool = False,
-        **kwargs
+        **kwargs,
     ):
         super().__init__(dist_sync_on_step=dist_sync_on_step, **kwargs)
 
-        self.n_obs_total = 1 if n_obs_total is None else n_obs_total
+        self._name = name
         self._mode = mode
+        self._interval = interval
 
         default_val = torch.tensor(0.0)
-        self.add_state("reconstruction_loss", default=default_val)
-        self.add_state("kl_local", default=default_val)
-        self.add_state("kl_global", default=default_val)
+        self.add_state("elbo_component", default=default_val)
         self.add_state("n_obs", default=default_val)
         self.add_state("n_batches", default=default_val)
 
     @property
     def mode(self):
         return self._mode
 
+    @property
+    def name(self):
+        return f"{self._name}_{self.mode}"
+
+    @name.setter
+    def name(self, new_name):
+        self._name = new_name
+
+    @property
+    def interval(self):
+        return self._interval
+
+    def get_intervals_recorded(self):
+        if self.interval == "obs":
+            return self.n_obs
+        elif self.interval == "batch":
+            return self.n_batches
+        raise ValueError(f"Unrecognized interval: {self.interval}.")
+
     def update(
         self,
-        reconstruction_loss_sum: torch.Tensor,
-        kl_local_sum: torch.Tensor,
-        kl_global: torch.Tensor,
-        n_obs_minibatch: int,
+        **kwargs,
     ):
         """Updates all metrics."""
```

Comment:
might be good to explain somehwere that in a metric collection that arguments get passed to each individual metric